kong/runloop/plugin_servers/init.lua

```diff
@@ -15,6 +17,75 @@ local rpc_notifications = {}
 --- currently running plugin instances
 local running_instances = {}
 
+local exposed_api = {
+  kong = kong,
+
+  ["kong.log.serialize"] = function()
+    local saved = save_for_later[coroutine.running()]
```

Comment:
We should be able to cache `coroutine.running` as well to speed up the hot path.

Reply:
it's used to differentiate the request that is being handled.  the problem is that the `log` phase is handled in a timer context, after the request object itself has been released, so we have to associate those values with the timer instance.  I remember looking for some timer ID but didn't find anything usable.  I'll check again.